Using an SDK that comes from a PackageReference in a WPF application is broken

* .NET Core Version: 3.0preview 7 + 3.0rc1
* Windows version: Several Win10 versions and Win11 (ie doesn't matter)
* Does the bug reproduce also in WPF for .NET Framework 4.8?: No
* Is this bug related specifically to tooling in Visual Studio (e.g. XAML Designer, Code editing, etc...)? If yes, please file the issue via the instructions [here](https://docs.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2019).
No? It is introduced with .NET 6 and/or VS2022 installer, but will even affect VS2019/.NET5 installs once installed
 
 **Problem description:**
Using an SDK that comes from a PackageReference (MSBuild.Sdk.Extras) in a WPF application is broken with WPF PackageReference support enabled. 
Setting `IncludePackageReferencesDuringMarkupCompilation` to `false` works around the issue (but comes with other downsides).
On a PC that only has VS2019 installed, the problem doesn't reproduce. If you install VS2022p3.1, the problem does reproduce, even from a VS2019 command prompt.
The problem seems to occur just by installing the .NET 6 SDK, and will break even .NET 5 and .NET Framework builds.

 **Actual behavior:** <!-- callstack for crashes / exceptions -->
 > error MSB4236: The SDK 'MSBuild.Sdk.Extras/3.0.23' specified could not be found.

 **Expected behavior:**
 Build completes.

 **Minimal repro:**
From a VS2019 or VS2022 prompt run:
```
    git clone https://github.com/Esri/arcgis-toolkit-dotnet
    cd arcgis-toolkit-dotnet\src\Toolkit\Toolkit
    msbuild /restore /t:Build Esri.ArcGISRuntime.Toolkit.csproj
```

--- a/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs
+++ b/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs
@@ -774,6 +774,17 @@ namespace Microsoft.Build.Tasks.Windows
                     var sdkValue = xmlAttribute.Value;
                     root.Attributes.Remove(xmlAttribute);
 
+                    // When a version is specified in the Sdk attribute,
+                    // split the Sdk attribute in Sdk and Version.
+                    // The split is required when using explicit import.
+                    string sdkVersionValue = null;
+                    int versionSeparatorIndex = sdkValue.IndexOf("/");
+                    if (versionSeparatorIndex != -1)
+                    {
+                        sdkVersionValue = sdkValue.Substring(versionSeparatorIndex + 1);
+                        sdkValue = sdkValue.Substring(0, versionSeparatorIndex);
+                    }
+
                     //
                     // Add explicit top import
                     //
@@ -782,8 +793,18 @@ namespace Microsoft.Build.Tasks.Windows
                     XmlNode nodeImportProps = xmlProjectDoc.CreateElement("Import", root.NamespaceURI);
                     XmlAttribute projectAttribute = xmlProjectDoc.CreateAttribute("Project");
                     projectAttribute.Value = "Sdk.props";
+                    XmlAttribute sdkAttributeProps = xmlProjectDoc.CreateAttribute("Sdk");
+                    sdkAttributeProps.Value = sdkValue;
                     nodeImportProps.Attributes.Append(projectAttribute);
-                    nodeImportProps.Attributes.Append(xmlAttribute);
+                    nodeImportProps.Attributes.Append(sdkAttributeProps);
+
+                    // Add Sdk version when specified.
+                    if (sdkVersionValue != null)
+                    {
+                        XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("Version");
+                        sdkVersionAttributeProps.Value = sdkVersionValue;
+                        nodeImportProps.Attributes.Append(sdkVersionAttributeProps);
+                    }
 
                     // Prepend this Import to the root of the XML document
                     root.PrependChild(nodeImportProps);
@@ -800,6 +821,14 @@ namespace Microsoft.Build.Tasks.Windows
                     projectAttribute3.Value = sdkValue;
                     nodeImportTargets.Attributes.Append(projectAttribute2);
                     nodeImportTargets.Attributes.Append(projectAttribute3);
+
+                    // Add Sdk version when specified.
+                    if (sdkVersionValue != null)
+                    {
+                        XmlAttribute sdkVersionAttributeTargets = xmlProjectDoc.CreateAttribute("Version");
+                        sdkVersionAttributeTargets.Value = sdkVersionValue;
+                        nodeImportTargets.Attributes.Append(sdkVersionAttributeTargets);
+                    }
 
                     // Append this Import to the end of the XML document
                     root.AppendChild(nodeImportTargets);

--- a/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs
+++ b/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs
@@ -23,6 +23,7 @@ using System.Collections.Generic;
 
 using System.Globalization;
 using System.Diagnostics;
+using System.Linq;
 using System.Reflection;
 using System.Resources;
 using System.Xml;
@@ -761,50 +762,44 @@ namespace Microsoft.Build.Tasks.Windows
             }
 
             XmlNode root = xmlProjectDoc.DocumentElement;
-          
+
             for (int i = 0; i < root.Attributes.Count; i++)
             {
                 XmlAttribute xmlAttribute = root.Attributes[i] as XmlAttribute;
 
                 if (xmlAttribute.Name.Equals("Sdk", StringComparison.OrdinalIgnoreCase))
                 {
-                    // <Project Sdk="Microsoft.NET.Sdk">
+                    //  <Project Sdk="Microsoft.NET.Sdk">
+                    //  <Project Sdk="My.Custom.Sdk/1.0.0" />
+                    //  <Project Sdk="My.Custom.Sdk/min=1.0.0" />
 
                     // Remove Sdk attribute
                     var sdkValue = xmlAttribute.Value;
                     root.Attributes.Remove(xmlAttribute);
 
-                    // When a version is specified in the Sdk attribute,
-                    // split the Sdk attribute in Sdk and Version.
-                    // The split is required when using explicit import.
-                    string sdkVersionValue = null;
-                    int versionSeparatorIndex = sdkValue.IndexOf("/");
-                    if (versionSeparatorIndex != -1)
+                    string minimumVersionAttributeValue = null;
+                    string versionAttributeValue = null;
+                    var sdkParts = sdkValue.Split('/').Select(i => i.Trim()).ToArray();
+                    sdkValue = sdkParts[0];
+
+                    if (sdkParts.Length == 2)
                     {
-                        sdkVersionValue = sdkValue.Substring(versionSeparatorIndex + 1);
-                        sdkValue = sdkValue.Substring(0, versionSeparatorIndex);
+                        if (sdkParts[1].StartsWith("min=", StringComparison.OrdinalIgnoreCase))
+                        {
+                            minimumVersionAttributeValue = sdkParts[1].Substring(4);
+                        }
+                        else
+                        {
+                            versionAttributeValue = sdkParts[1];
+                        }
                     }
 
                     //
                     // Add explicit top import
                     //
-                    //  <Import Project = "Sdk.props" Sdk="Microsoft.NET.Sdk" />
+                    //  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
                     //
-                    XmlNode nodeImportProps = xmlProjectDoc.CreateElement("Import", root.NamespaceURI);
-                    XmlAttribute projectAttribute = xmlProjectDoc.CreateAttribute("Project");
-                    projectAttribute.Value = "Sdk.props";
-                    XmlAttribute sdkAttributeProps = xmlProjectDoc.CreateAttribute("Sdk");
-                    sdkAttributeProps.Value = sdkValue;
-                    nodeImportProps.Attributes.Append(projectAttribute);
-                    nodeImportProps.Attributes.Append(sdkAttributeProps);
-
-                    // Add Sdk version when specified.
-                    if (sdkVersionValue != null)
-                    {
-                        XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("Version");
-                        sdkVersionAttributeProps.Value = sdkVersionValue;
-                        nodeImportProps.Attributes.Append(sdkVersionAttributeProps);
-                    }
+                    XmlNode nodeImportProps = CreateImportProjectSdkNode(xmlProjectDoc, "Sdk.props", sdkValue, versionAttributeValue, minimumVersionAttributeValue);
 
                     // Prepend this Import to the root of the XML document
                     root.PrependChild(nodeImportProps);
@@ -812,28 +807,49 @@ namespace Microsoft.Build.Tasks.Windows
                     //
                     // Add explicit bottom import
                     //
-                    //  <Import Project = "Sdk.targets" Sdk="Microsoft.NET.Sdk" 
-                    //                
-                    XmlNode nodeImportTargets = xmlProjectDoc.CreateElement("Import", root.NamespaceURI);
-                    XmlAttribute projectAttribute2 = xmlProjectDoc.CreateAttribute("Project");
-                    projectAttribute2.Value = "Sdk.targets";
-                    XmlAttribute projectAttribute3 = xmlProjectDoc.CreateAttribute("Sdk");
-                    projectAttribute3.Value = sdkValue;
-                    nodeImportTargets.Attributes.Append(projectAttribute2);
-                    nodeImportTargets.Attributes.Append(projectAttribute3);
-
-                    // Add Sdk version when specified.
-                    if (sdkVersionValue != null)
-                    {
-                        XmlAttribute sdkVersionAttributeTargets = xmlProjectDoc.CreateAttribute("Version");
-                        sdkVersionAttributeTargets.Value = sdkVersionValue;
-                        nodeImportTargets.Attributes.Append(sdkVersionAttributeTargets);
-                    }
+                    //  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" 
+                    //
+                    XmlNode nodeImportTargets = CreateImportProjectSdkNode(xmlProjectDoc,"Sdk.targets", sdkValue, versionAttributeValue, minimumVersionAttributeValue);
 
                     // Append this Import to the end of the XML document
                     root.AppendChild(nodeImportTargets);
                 }
             }
+        }
+
+        // Creates an XmlNode that contains an Import Project element
+        //
+        //  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+        static XmlNode CreateImportProjectSdkNode(
+            XmlDocument xmlProjectDoc, 
+            string projectAttributeValue, 
+            string sdkAttributeValue,
+            string versionAttributeValue,
+            string minimumVersionAttributeValue)
+        {
+            XmlNode nodeImport = xmlProjectDoc.CreateElement("Import", xmlProjectDoc.DocumentElement.NamespaceURI);
+            XmlAttribute projectAttribute = xmlProjectDoc.CreateAttribute("Project");
+            projectAttribute.Value = projectAttributeValue;
+            XmlAttribute sdkAttributeProps = xmlProjectDoc.CreateAttribute("Sdk");
+            sdkAttributeProps.Value = sdkAttributeValue;
+            nodeImport.Attributes.Append(projectAttribute);
+            nodeImport.Attributes.Append(sdkAttributeProps);
+
+            if (!string.IsNullOrEmpty(versionAttributeValue))
+            {
+                XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("Version");
+                sdkVersionAttributeProps.Value = versionAttributeValue;
+                nodeImport.Attributes.Append(sdkVersionAttributeProps);
+            }
+
+            if (!string.IsNullOrEmpty(minimumVersionAttributeValue))
+            {
+                XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("MinimumVersion");
+                sdkVersionAttributeProps.Value = minimumVersionAttributeValue;
+                nodeImport.Attributes.Append(sdkVersionAttributeProps);
+            }
+
+            return nodeImport;
         }
 
         #endregion Private Methods

--- a/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs
+++ b/src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs
@@ -23,7 +23,6 @@ using System.Collections.Generic;
 
 using System.Globalization;
 using System.Diagnostics;
-using System.Linq;
 using System.Reflection;
 using System.Resources;
 using System.Xml;
@@ -770,36 +769,25 @@ namespace Microsoft.Build.Tasks.Windows
                 if (xmlAttribute.Name.Equals("Sdk", StringComparison.OrdinalIgnoreCase))
                 {
                     //  <Project Sdk="Microsoft.NET.Sdk">
-                    //  <Project Sdk="My.Custom.Sdk/1.0.0" />
-                    //  <Project Sdk="My.Custom.Sdk/min=1.0.0" />
+                    //  <Project Sdk="My.Custom.Sdk/1.0.0">
+                    //  <Project Sdk="My.Custom.Sdk/min=1.0.0">
+
+                    string sdkValue = xmlAttribute.Value;
+
+                    if (!SdkReference.TryParse(sdkValue, out SdkReference sdkReference))
+                        return;
 
                     // Remove Sdk attribute
-                    var sdkValue = xmlAttribute.Value;
                     root.Attributes.Remove(xmlAttribute);
-
-                    string minimumVersionAttributeValue = null;
-                    string versionAttributeValue = null;
-                    var sdkParts = sdkValue.Split('/').Select(i => i.Trim()).ToArray();
-                    sdkValue = sdkParts[0];
-
-                    if (sdkParts.Length == 2)
-                    {
-                        if (sdkParts[1].StartsWith("min=", StringComparison.OrdinalIgnoreCase))
-                        {
-                            minimumVersionAttributeValue = sdkParts[1].Substring(4);
-                        }
-                        else
-                        {
-                            versionAttributeValue = sdkParts[1];
-                        }
-                    }
 
                     //
                     // Add explicit top import
                     //
                     //  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+                    //  <Import Project="Sdk.props" Sdk="My.Custom.Sdk" Version="1.0.0" />
+                    //  <Import Project="Sdk.props" Sdk="My.Custom.Sdk" MinimumVersion="1.0.0" />
                     //
-                    XmlNode nodeImportProps = CreateImportProjectSdkNode(xmlProjectDoc, "Sdk.props", sdkValue, versionAttributeValue, minimumVersionAttributeValue);
+                    XmlNode nodeImportProps = CreateImportProjectSdkNode(xmlProjectDoc, "Sdk.props", sdkReference);
 
                     // Prepend this Import to the root of the XML document
                     root.PrependChild(nodeImportProps);
@@ -807,9 +795,11 @@ namespace Microsoft.Build.Tasks.Windows
                     //
                     // Add explicit bottom import
                     //
-                    //  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" 
+                    //  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+                    //  <Import Project="Sdk.targets" Sdk="My.Custom.Sdk" Version="1.0.0" />
+                    //  <Import Project="Sdk.targets" Sdk="My.Custom.Sdk" MinimumVersion="1.0.0" />
                     //
-                    XmlNode nodeImportTargets = CreateImportProjectSdkNode(xmlProjectDoc,"Sdk.targets", sdkValue, versionAttributeValue, minimumVersionAttributeValue);
+                    XmlNode nodeImportTargets = CreateImportProjectSdkNode(xmlProjectDoc, "Sdk.targets", sdkReference);
 
                     // Append this Import to the end of the XML document
                     root.AppendChild(nodeImportTargets);
@@ -820,32 +810,27 @@ namespace Microsoft.Build.Tasks.Windows
         // Creates an XmlNode that contains an Import Project element
         //
         //  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
-        static XmlNode CreateImportProjectSdkNode(
-            XmlDocument xmlProjectDoc, 
-            string projectAttributeValue, 
-            string sdkAttributeValue,
-            string versionAttributeValue,
-            string minimumVersionAttributeValue)
+        static XmlNode CreateImportProjectSdkNode(XmlDocument xmlProjectDoc, string projectAttributeValue, SdkReference sdkReference)
         {
             XmlNode nodeImport = xmlProjectDoc.CreateElement("Import", xmlProjectDoc.DocumentElement.NamespaceURI);
             XmlAttribute projectAttribute = xmlProjectDoc.CreateAttribute("Project");
             projectAttribute.Value = projectAttributeValue;
             XmlAttribute sdkAttributeProps = xmlProjectDoc.CreateAttribute("Sdk");
-            sdkAttributeProps.Value = sdkAttributeValue;
+            sdkAttributeProps.Value = sdkReference.Name;
             nodeImport.Attributes.Append(projectAttribute);
             nodeImport.Attributes.Append(sdkAttributeProps);
 
-            if (!string.IsNullOrEmpty(versionAttributeValue))
+            if (!string.IsNullOrEmpty(sdkReference.Version))
             {
                 XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("Version");
-                sdkVersionAttributeProps.Value = versionAttributeValue;
+                sdkVersionAttributeProps.Value = sdkReference.Version;
                 nodeImport.Attributes.Append(sdkVersionAttributeProps);
             }
 
-            if (!string.IsNullOrEmpty(minimumVersionAttributeValue))
+            if (!string.IsNullOrEmpty(sdkReference.MinimumVersion))
             {
                 XmlAttribute sdkVersionAttributeProps = xmlProjectDoc.CreateAttribute("MinimumVersion");
-                sdkVersionAttributeProps.Value = minimumVersionAttributeValue;
+                sdkVersionAttributeProps.Value = sdkReference.MinimumVersion;
                 nodeImport.Attributes.Append(sdkVersionAttributeProps);
             }
 

